src/treelearner/serial_tree_learner.cpp

```diff
@@ -711,20 +711,29 @@ void SerialTreeLearner::ComputeBestSplitForFeature(
     FeatureHistogram* histogram_array_, int feature_index, int real_fidx,
     bool is_feature_used, int num_data, const LeafSplits* leaf_splits,
     SplitInfo* best_split) {
+
+  bool is_feature_numerical = train_data_->FeatureBinMapper(feature_index)
+                                  ->bin_type() == BinType::NumericalBin;
+  if (is_feature_numerical & !config_->monotone_constraints.empty()) {
+    constraints_->RecomputeConstraintsIfNeeded(
+        constraints_.get(), feature_index, ~(leaf_splits->leaf_index()),
+        train_data_->FeatureNumBin(feature_index));
+  }
+  
```

Comment:
@CharlesAuguste Could you please fix lint issue which causes CI test to fail?
```
src/treelearner/serial_tree_learner.cpp:722:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
```

Reply:
@StrikerRUS Sorry about that I missed that line and didn't understand why the job failed. It is al good now!

Reply:
No problem, thank you!